mypy/checkexpr.py

```diff
@@ -250,6 +250,11 @@ def check_call(self, callee: Type, args: List[Node],
             self.check_argument_types(arg_types, arg_kinds, callee,
                                       formal_to_actual, context,
                                       messages=arg_messages)
+
+            ret_val_is_type_obj = is_equivalent(callee.ret_type, self.named_type('builtins.type'))
```

Comment:
That's a pretty expensive call, and this case rarely triggers, so the time is usually wasted. I suggest testing the simpler conditions first and doing this call last.